mmdet/core/results/results.py

```diff
@@ -0,0 +1,346 @@
+
+    __delitem__ = __delattr__
+
+    # Tensor-like methods
+    def to(self, *args, **kwargs):
+        """Apply same name function to all tensors in results field."""
+        new_instance = self.new_results()
+        for k, v in self._results_field.items():
+            if isinstance(v, torch.Tensor):
+                new_instance[k] = v.to(*args, **kwargs)
+        return new_instance
+
+    # Tensor-like methods
+    def cpu(self):
+        """Apply same name function to all tensors in results field."""
+        new_instance = self.new_results()
+        for k, v in self._results_field.items():
+            if isinstance(v, torch.Tensor):
+                new_instance[k] = v.cpu()
+        return new_instance
+
+    # Tensor-like methods
+    def cuda(self):
+        """Apply same name function to all tensors in results field."""
+        new_instance = self.new_results()
+        for k, v in self._results_field.items():
+            if isinstance(v, torch.Tensor):
+                new_instance[k] = v.cuda()
+        return new_instance
+
+    # Tensor-like methods
+    def detach(self):
+        """Apply same name function to all tensors in results field."""
+        new_instance = self.new_results()
+        for k, v in self._results_field.items():
+            if isinstance(v, torch.Tensor):
+                new_instance[k] = v.detach()
+        return new_instance
+
+    # Tensor-like methods
+    def numpy(self):
+        """Apply same name function to all tensors in results field."""
+        new_instance = self.new_results()
+        for k, v in self._results_field.items():
+            if isinstance(v, torch.Tensor):
+                new_instance[k] = v.cpu().numpy()
+        return new_instance
+
+    def __nice__(self):
+        repr = '\n   META INFORMATION \n'
+        for k, v in self._meta_info_field.items():
+            repr += f'{k}: {v} \n'
+        repr += '\n   PREDICTIONS \n'
+        for k, v in self._results_field.items():
+            if isinstance(v, (torch.Tensor, np.ndarray)):
+                repr += f'shape of {k}: {v.shape} \n'
+            else:
+                repr += f'{k}: {v} \n'
+        return repr
+
+    def __copy__(self):
+        # Due to the overloading of  `__getattr__`,
+        # have to customize the `__copy__`.
+        new_instance = self.__class__()
+        new_instance.__dict__.update(self.__dict__)
+        return new_instance
+
+    def __deepcopy__(self, memo=None):
+        # Due to the overloading of  `__getattr__`,
+        # have to customize the `__deepcopy__`.
+        new_instance = self.__class__()
+        for k, v in self.__dict__.items():
+            if isinstance(v, dict):
+                new_instance[k] = copy.deepcopy(v)
+        return new_instance
+
+
+class InstanceResults(Results):
+
+    def __len__(self):
+        for v in self._results_field.values():
+            return v.__len__()
+        raise NotImplementedError('This is an empty instance.')
```

Comment:
For empty instance, should we raise Error for that? May be it will be enriched later?